src/java.base/unix/native/libjli/java_md.c

```diff
@@ -313,31 +313,34 @@ CreateExecutionEnvironment(int *pargc, char ***pargv,
     /* Compute/set the name of the executable */
     SetExecname(*pargv);
 
-    /* Check to see if the jvmpath exists */
-    /* Find out where the JRE is that we will be using. */
-    if (!GetJREPath(jrepath, so_jrepath, JNI_FALSE)) {
-        JLI_ReportErrorMessage(JRE_ERROR1);
-        exit(2);
-    }
-    JLI_Snprintf(jvmcfg, so_jvmcfg, "%s%slib%sjvm.cfg",
-            jrepath, FILESEP, FILESEP);
-    /* Find the specified JVM type */
-    if (ReadKnownVMs(jvmcfg, JNI_FALSE) < 1) {
-        JLI_ReportErrorMessage(CFG_ERROR7);
-        exit(1);
-    }
+    if (!JLI_IsStaticallyLinked()) {
```

Comment:
Any reason this is diverted from the change in hermetic Java branch, https://github.com/openjdk/leyden/blob/c1c5fc686c1452550e1b3663a320fba652248505/src/java.base/unix/native/libjli/java_md.c#L300? I think the setenv part below is not needed for static/hermetic support either. There is no $JRE/lib with a single executable image. All natives are statically linked with the executable image.

Reply:
I do not recall my reason for deviating from the hermetic Java branch. I have double-checked the code, and I agree with you that the setenv part is not needed for static libraries, since all it does is trying to use `LD_LIBRARY_PATH` properly to load the dynamic libraries, so the short-cut solution from the hermetic-java-runtime branch would suffice.

However, the `SetExecname()` call must be done, otherwise the re-execution of the binary as done on macOS (to be able to release the main thread to the GUI handler) will fail. This was probably the reason I started changing this function at all. But after the `SetExecname()` call, we can exit the rest if we are on a static build.